acceptance-tests/src/main/java/io/blueocean/ath/SmartWebElement.java

```diff
@@ -0,0 +1,284 @@
+package io.blueocean.ath;
+
+import com.google.common.base.Preconditions;
+import org.openqa.selenium.*;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.FluentWait;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Wrapper around an underlying WebDriver that
+ * consistently handles waits automatically.
+ *
+ * Accepts expressions for css and xpath, if the provided lookup starts with a /, XPath is used
+ */
+public class SmartWebElement implements WebElement {
+    private WebDriver driver;
+    protected String expr;
+    protected By by;
+
+    public SmartWebElement(WebDriver driver, String expr) {
+        this.driver = driver;
+        this.expr = expr;
+        if (expr.startsWith("/")) {
+            by = By.xpath(expr);
+        } else {
+            by = By.cssSelector(expr);
+        }
+    }
+
+    public SmartWebElement(WebDriver driver, By by) {
+        this.driver = driver;
+        this.expr = by.toString();
+        this.by = by;
+    }
+
+    /**
+     * Gets a WebDriver instance
+     * @return from threadlocal
+     */
+    protected WebDriver getDriver() {
+        return driver;
+    }
+
+    /**
+     * Gets elements
+     * @return the elements found
+     */
+    public List<WebElement> getElements() {
+        return new FluentWait<>(getDriver())
+            .pollingEvery(100, TimeUnit.MILLISECONDS)
+            .withTimeout(WaitUtil.DEFAULT_TIMEOUT, TimeUnit.MILLISECONDS)
+            .ignoring(NoSuchElementException.class)
+            .until(ExpectedConditions.numberOfElementsToBeMoreThan(by, 0));
+    }
+
+    /**
+     * Gets the first matching element
+     * @return see description
+     */
+    public WebElement getElement() {
+        return getElements().iterator().next();
+    }
+
+    /**
+     * Iterates over all found elements with the given function
+     * @param fn to perform on the elements
+     */
+    public void forEach(java.util.function.Consumer<WebElement> fn) {
+        for (WebElement e : getElements()) {
+            fn.accept(e);
+        }
+    }
+
+    public boolean isVisible() {
+        return getElement().isDisplayed();
+    }
+
+    @Override
+    public void click() {
+        forEach(e -> e.click());
```

Comment:
Do want to support clicking multiple elements? It seems that it could have unintended consequences if the selector is too loose. Perhaps warn if multiple elements are matched?

We should bake in the retry logic from `WaitUtil.click` here I believe.